src/components/application_manager/test/hmi_capabilities_test.cc

```diff
@@ -1129,6 +1120,155 @@ TEST_F(HMICapabilitiesTest,
+      .WillByDefault(ReturnRef(mock_dispatcher));
+  ON_CALL(mock_app_mngr, get_settings())
+      .WillByDefault(ReturnRef(mock_application_manager_settings));
+  ON_CALL(mock_application_manager_settings,
+              hmi_capabilities_cache_file_name())
+      .WillByDefault(ReturnRef(file_cache_name_));
+
+  auto hmi_capabilities =
+      std::make_shared<HMICapabilitiesForTesting>(mock_app_mngr);
+
+  file_system::CreateFile(file_cache_name_);
+  EXPECT_TRUE(file_system::FileExists(file_cache_name_));
+
+  const std::vector<uint8_t> binary_data_to_save(content_to_save.begin(),
+                                                 content_to_save.end());
+  file_system::Write(file_cache_name_, binary_data_to_save);
+
+  EXPECT_TRUE(hmi_capabilities->SaveCachedCapabilitiesToFile(
+      hmi_interface::tts, sections_to_update, schema));
+
+  std::string content_after_update;
+  EXPECT_TRUE(file_system::ReadFile(file_cache_name_, content_after_update));
+  EXPECT_NE(content_after_update, content_to_save);
+}
+
+TEST_F(HMICapabilitiesTest, PrepareJsonValueForSaving_Success) {
+  MockApplicationManager mock_app_mngr;
+  event_engine_test::MockEventDispatcher mock_dispatcher;
+  MockApplicationManagerSettings mock_application_manager_settings;
+  const smart_objects::CSmartSchema schema;
+
+  const std::vector<std::string> sections_to_update{
+      hmi_response::display_capabilities,
+      hmi_response::hmi_zone_capabilities,
+      hmi_response::soft_button_capabilities,
+      strings::audio_pass_thru_capabilities,
+      strings::hmi_capabilities,
+      strings::system_capabilities,
+      hmi_response::languages};
+
+  const std::vector<std::string> interfaces_name{
+      hmi_interface::tts,
+      hmi_interface::vr,
+      hmi_interface::ui,
+      hmi_interface::buttons,
+      hmi_interface::vehicle_info,
+      hmi_interface::rc,
+      hmi_response::speech_capabilities,
+      hmi_response::prerecorded_speech_capabilities,
+      hmi_response::button_capabilities,
+      hmi_response::preset_bank_capabilities};
+
+  const std::string content_to_save = "{\"field\" : \"value\" }";
+  const std::vector<uint8_t> binary_data_to_save(content_to_save.begin(),
+                                                 content_to_save.end());
+
+  ON_CALL(mock_app_mngr, event_dispatcher())
+      .WillByDefault(ReturnRef(mock_dispatcher));
+  ON_CALL(mock_app_mngr, get_settings())
+      .WillByDefault(ReturnRef(mock_application_manager_settings));
+  ON_CALL(mock_application_manager_settings,
+              hmi_capabilities_cache_file_name())
+      .WillByDefault(ReturnRef(file_cache_name_));
+
+  auto hmi_capabilities =
+      std::make_shared<HMICapabilitiesForTesting>(mock_app_mngr);
+
+  file_system::CreateFile(file_cache_name_);
+  EXPECT_TRUE(file_system::FileExists(file_cache_name_));
+
+  file_system::Write(file_cache_name_, binary_data_to_save);
+
+  for (size_t i = 0; i < interfaces_name.size(); i++) {
```

Comment:
@ydementieiev 
I always recommend to use `++i`. It’s a useful habit to get into.

` for(int i=0; i<10; ++i) { }`

Most of the time it is an integer, and it has no benefit. But when it is an iterator, perhaps a complex one, it avoids a temporary copy.

Reply:
@AByzhynar [changed](https://github.com/smartdevicelink/sdl_core/pull/3300/commits/5d1e1c86c015bb592f9cd47f37c4f780459bcc9d#diff-c1d0a5558de3ab3e31ce8813484ebefeR1249) thank you